### PR TITLE
Creating Test for the Cloud Storage Library

### PR DESCRIPTION
**Pull request by mrjwashiv**

### CODE REVIEW

File: project/src/test/java/com/google/sps/cloudstoragetests/GcsFilePathTest.java [hunk 1]
```diff
@@ -54,16 +54,16 @@ public void setUp() {
     }
     
     @Test
-    public void EmptyMapForGcsFilePath() {
-        String actual = getGcsFilePath(NO_FILES);
+    public void emptyMapForGetGcsFilePath() {
```
Comment (OlagunjuPhillips) [inline project/src/test/java/com/google/sps/cloudstoragetests/GcsFilePathTest.java:61]: I think these all look good, but can you add comments to each tests to explain what each test is for? I think that would be good.

File: project/src/test/java/com/google/sps/cloudstoragetests/GcsFilePathTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import com.google.sps.data.CloudStorageLibrary;
+
+@RunWith(JUnit4.class)
+public final class GcsFilePathTest {
+    //Empty Map of files
+    private static final Map<String, List<FileInfo>> NO_FILES = Collections.emptyMap();
+
+    //Defining sample files for testing
+    private static final Date date = new Date();
+    private static final Long size = 27262L;
```
Comment (mariabajzek) [inline project/src/test/java/com/google/sps/cloudstoragetests/GcsFilePathTest.java:25]: Is this specific size relevant? If so, it might be nice to have a comment explaining why you chose this size.

However, if it's just a random number for testing purposes, then I don't think you need to explain it.

File: project/src/test/java/com/google/sps/cloudstoragetests/GcsFilePathTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        String expected = "";
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    //Test what happens when you pass in a map that contains files
+    @Test
+    public void getGcsFilePathIsSet() {
+        String actual = CloudStorageLibrary.getGcsFilePath(FILES);
+        String expected = "/gs/myBucket/file1";
```
Comment (mariabajzek) [inline project/src/test/java/com/google/sps/cloudstoragetests/GcsFilePathTest.java:72]: It's not obvious to me why the path for "file1" is expected here. It would be helpful to add a comment about why this is the expected output (rather than a different file or just the path to the bucket).

File: project/src/test/java/com/google/sps/cloudstoragetests/ServingFileUrlTest.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    //Function that test what happens if you pass in a empty string
+    @Test
+    public void servingEmptyGcsFileName() {
+        String actual = CloudStorageLibrary.getServingFileUrl("");
+        String expected = "";
+
+        Assert.assertEquals(expected, actual);
+    }
+
+    //Test to make sure the function works properly if you pass in a file name
```
Comment (mariabajzek) [inline project/src/test/java/com/google/sps/cloudstoragetests/ServingFileUrlTest.java:86]: One test you could add is what happens if you pass in a file name that doesn't exist.

File: project/src/test/java/com/google/sps/cloudstoragetests/UploadedFileUrlTest.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Before
+    public void setUp() {
+        when(MOCK_BLOBSTORE.createGsBlobKey(GCS_FILE_NAME)).thenReturn(MOCK_BLOBKEY);
+        when(MOCK_BLOBKEY.getKeyString()).thenReturn("123456789");
+
+        BLOBKEY = MOCK_BLOBSTORE.createGsBlobKey(GCS_FILE_NAME).getKeyString();
+    }
+    
+    //Test to show what happens when you pass in a null value
```
Comment (mariabajzek) [inline project/src/test/java/com/google/sps/cloudstoragetests/UploadedFileUrlTest.java]: You could also test what happens if you pass in a null Blobstore instance.

**Review by mariabajzek — CHANGES_REQUESTED** [review]

**Review by mariabajzek — COMMENTED** [review]
Meant to add that this is a nice set of tests! I just left a few suggestions.

File: project/src/test/java/com/google/sps/cloudstoragetests/GcsFilePathTest.java (hunk 2 above)
Comment (mrjwashiv) [reply project/src/test/java/com/google/sps/cloudstoragetests/GcsFilePathTest.java:25]: Initially I had it as the typical size of a file, but I was struggling on how to initialize a long, and I finally found out that it needs an L at the end. But through all the changes the number did not stay the same. I can change it though.

**Review by mariabajzek — APPROVED** [review]
Looks good. Glad you were able to figure this out!
